packages/cardano-services/src/Utxo/DbSyncUtxoProvider/mappers.ts

```diff
@@ -0,0 +1,50 @@
+import { Cardano } from '@cardano-sdk/core';
+import { UtxoModel } from './types';
+import { generateAssetId, generateUtxoId } from './util';
+
+/**
+ * Transform DB results into indexed core UTxO
+ *
+ * @param {UtxoModel[]} utxosModels  UTxO query rows
+ * @returns {Cardano.Utxo[]} an array of core UTxO objects
+ */
+export const utxosToCore = (utxosModels: UtxoModel[]): Cardano.Utxo[] => {
+  const utxosMap = utxosModels.reduce((utxos, current) => {
+    const utxoId = generateUtxoId(current.tx_id, current.index);
+    const utxo = utxos.get(utxoId);
+    if (utxo) {
```

Comment:
The idea is to index the reduce object by UTxO. Since we have one row (or `UtxoModel`) per each asset (due to the SQL query), we can have multiple rows with the same UTxO. It is in these cases where we must check if the UTxO already exists and modify its assets.

Reply:
https://github.com/input-output-hk/cardano-js-sdk/pull/240#discussion_r873826884